### ISSUE TITLE
Generating a resource with no entity name throws an error.

### ISSUE THREAD

**Issue by kwight** [ember-cli/ember-cli#4665]
Generating a resource with no entity name throws an error and stack trace:

```
Arthur:hummingbird kwight$ ember generate resource
version: 1.13.8-master-1ca9464dac
Cannot read property 'toLowerCase' of undefined
TypeError: Cannot read property 'toLowerCase' of undefined
    at Object.inflector._apply_rules (/Users/kwight/Sites/ember-cli/node_modules/inflection/lib/inflection.js:579:52)
    at Object.inflector.singularize (/Users/kwight/Sites/ember-cli/node_modules/inflection/lib/inflection.js:677:24)
    at Class.module.exports._process (/Users/kwight/Sites/ember-cli/blueprints/resource/index.js:55:26)
    at Class.module.exports.install (/Users/kwight/Sites/ember-cli/blueprints/resource/index.js:14:17)
    at Class.module.exports.Task.extend.run (/Users/kwight/Sites/ember-cli/lib/tasks/generate-from-blueprint.js:60:49)
    at Class.module.exports.Command.extend.run (/Users/kwight/Sites/ember-cli/lib/commands/generate.js:80:17)
    at Class.<anonymous> (/Users/kwight/Sites/ember-cli/lib/models/command.js:146:17)
    at lib$rsvp$$internal$$tryCatch (/Users/kwight/Sites/ember-cli/node_modules/rsvp/dist/rsvp.js:493:16)
    at lib$rsvp$$internal$$invokeCallback (/Users/kwight/Sites/ember-cli/node_modules/rsvp/dist/rsvp.js:505:17)
    at lib$rsvp$$internal$$publish (/Users/kwight/Sites/ember-cli/node_modules/rsvp/dist/rsvp.js:476:11)
```

instead of the helpful descriptive error in `lib/models/blueprint.js` for, say, a route:

```
Arthur:hummingbird kwight$ ember generate route
version: 1.13.8-master-1ca9464dac
installing route
The `ember generate` command requires an entity name to be specified. For more details, use `ember help`.
```

**Comment by kellyselden** [comment]
Fixed in #4667

**Comment by kellyselden** [comment]
Thanks for the report. Nice find.
